lib/model.js

```diff
@@ -2374,7 +2374,11 @@ class Model {
 
         return Promise.map(instances, instance =>
           instance.validate(validateOptions).catch(err => {
-            errors.push({record: instance, errors: err});
+            /**
+             * Simply extending the value of `err` would cause a breaking change as the `errors`
+             * property is expected on each item in the array of errors.
+             */
+            errors.push(new sequelizeErrors.RecordLinkedError(err, instance));
```

Comment:
Why not just push validation error

Reply:
It would cause a breaking change, a property of `record` and `errors` is expected in each item.

Reply:
I think we can simply do this

```js
err.record = instance;
errors.push(err);
```

In v5 we can remove `record` as `instance` property is already available in ValidationErrorItem